traffic_ops/traffic_ops_golang/deliveryservice/request/assign.go

```diff
@@ -223,12 +223,11 @@ func PutAssignment(w http.ResponseWriter, r *http.Request) {
 	}
 	var resp interface{}
 	if inf.Version.Major >= 4 {
-		if dsr.Requested != nil {
-			if dsr.Requested.LongDesc1 != nil || dsr.Requested.LongDesc2 != nil {
-				api.HandleErr(w, r, tx, http.StatusBadRequest, errors.New("the longDesc1 and longDesc2 fields are no longer supported in API 4.0 onwards"), nil)
-				return
-			}
+		if dsr.Requested != nil && dsr.Requested.LongDesc1 != nil || dsr.Requested.LongDesc2 != nil {
```

Comment:
```go
dsr.Requested != nil && dsr.Requested.LongDesc1 != nil || dsr.Requested.LongDesc2 != nil
```
is not equal to
```go
dsr.Requested != nil && (dsr.Requested.LongDesc1 != nil || dsr.Requested.LongDesc2 != nil)
```
for
* `dsr.Requested != nil` => false, `dsr.Requested.LongDesc1 != nil` => false, `dsr.Requested.LongDesc2 != nil` => true
* `dsr.Requested != nil` => false, `dsr.Requested.LongDesc1 != nil` => true, `dsr.Requested.LongDesc2 != nil` => true

Reply:
I was reading the if statement wrong, fixed now.